docs/transform.ipynb

```diff
@@ -116,61 +410,90 @@
+    "* A subclass of `cirq.Gateset`, which can be used for description and validation of gates supported by a specific target device / simulator.\n",
+    "* A bundle of pre/post transformers + an analytical decomposition method that can be used to rewrite an optimized equivalent circuit using gates from this target gateset. \n",
     "\n",
-    "You can create your own optimizers to transform and modify circuits to fit hardware, gate sets, or other requirements.  Optimizers can also be used to generate noise.  See [noise](noise.ipynb) for details.\n",
+    "Cirq has implementations for common target gatesets like `cirq.CZTargetGateset`, `cirq.SqrtIswapTargetGateset` etc.\n",
     "\n",
-    "You can do this by implementing the function `optimize_circuit`.\n",
     "\n",
-    "If your optimizer is a local optimizer and depends primarily on operator being examined, you can alternatively inherit `cirq.PointOptimizer` and implement the function `optimization_at(self, circuit, index, op)` that optimizes a single operation.\n",
+    "**`cirq.optimize_for_target_gateset`** is a transformer which compiles a given circuit for a `cirq.CompilationTargetGateset` gateset via the following steps:\n",
     "\n",
-    "Below is an example of implementing a simple `PointOptimizer` that removes measurements."
+    "1. Run all `gateset.preprocess_transformers`\n",
+    "2. Convert operations using built-in `cirq.decompose` + `gateset.decompose_to_target_gateset`.\n",
+    "3. Run all `gateset.postprocess_transformers`\n",
+    "\n",
+    "\n",
+    "The preprocess transformers often includes optimizations like merging connected components of 1/2 qubit unitaries into a single unitary matrix, which can then be replaced with an efficient analytical decomposition as part of step-2. \n",
+    "\n",
+    "The post-process transformers often includes cleanups and optimizations like dropping negligible operations, converting single qubit rotations into desired form, circuit alignments etc. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "-QFPCdW0qS3R"
+   },
+   "outputs": [],
+   "source": [
+    "context = cirq.TransformerContext(logger=cirq.TransformerLogger())\n",
+    "converted_circuit_optimized = cirq.optimize_for_target_gateset(circuit, gateset = cirq.CZTargetGateset(allow_partial_czs=True), context=context)\n",
+    "cirq.testing.assert_circuits_with_terminal_measurements_are_equivalent(circuit, converted_circuit_optimized)\n",
+    "print(f\"Circuit Depth Original: {len(circuit)}\", circuit, sep=\"\\n\")\n",
+    "print(f\"Circuit Depth Converted Custom: {len(converted_circuit)}\", converted_circuit, sep=\"\\n\")\n",
+    "print(f\"Circuit Depth Converted Optimized: {len(converted_circuit_optimized)}\", converted_circuit_optimized, sep=\"\\n\")"
```

Comment:
Nit: break up into multiple cells and walk through what's happening clearly.

Reply:
done